docs/contributing.md

```diff
@@ -0,0 +1,117 @@
+
+# Review Model
+
+Every project has its own code authoring and review culture which grows organically as the project matures and is a subject to change.
+Below is the description of the model we try to follow, with exceptions when the common sense dictates otherwise.
+
+## Review Types
+
+Review can be of either **architectural** or **technical** value, and often is of both.
+
+The PR author can specify, when requesting review, what kind of review they are asking for, from each reviewer, or even which area they'd like the reviewer to focus on (useful when distributing reviews).
+
+## Reviewer Role
+
+The reviewer is responsible for accepting a pull request only once they feel the current PR is ready to be merged even if their comments were not to be applied.
+
+The *accepted* can be set with pending review comments, if those comments don't affect whether the patch is ready to be merged (for example, they're stylistic suggestions).
+
+The should communicate the nature of their review comments - specifically, between the three types: *"blocking"*, *"suggestion"*, and *"optional"*.
+* **blocking** is when the reviewer considers the change to be unmergable and requires a new revision.
+* **suggestion** is for when the reviewer considers the change to be suboptimal, but usable, and wants to defer the decision to the PR author, while stating their opinion.
+* **optional** is for when the reviewer considers multiple options to be mostly comparable or tradeoffs, and wants to defer to the PR author for the final decision after bringing up a new option.
+
+## Social Contract
+
+`ICU4X` project focuses on a fairly hermetic domain of software internationalization, which requires prior knowledge of the domain.
+With that in mind, most engineers working on patch authoring and reviews are expected to be senior enough to be trusted with the quality of their additions to the code.
+
+For those reasons, we are primarily placing **trust** in pull request authors to write high quality, readable, tested, maintainable and well documented code.
+
+The role of the **reviewer** in such model is more conservative and is reduced to verification of the code from a particular angle with minimal impact on the pull request.
+Examples of such angle may be:
+
+* How the PR fits into the component public API
+* Memory management of the code in pull request
+* Test coverage, and sanity checks
+* Use of CLDR, Unicode, Rust and other best practices
+
+The pull request author is expected to evaluate what kind of review(s) they need to ensure the quality of their pull request.
+
+The role of the reviewer is not to *improve the code* or *educate* the PR author, but rather help the PR author accomplish their goal.
+
+In result most review comments should be in form of *suggestions* and *optionals*, and the reviewer should request another revision for review only if they feel the need to see it before they can evaluate if the PR is mergable.
```

Comment:
I don't know if I agree with this paragraph.  Reviewers should be encouraged to leave "required" suggestions if there's something wrong with relation to the bulleted list above, and 90% of the time, even with the best authors, there's going to be something worth fixing.

Reply:
Let's find a good sentence to capture that! I think your position is defensive against the other edge than mine, and we should be able to come up with a good middle.

I agree with you that reviewers should be encouraged to use `required` when the code is not ready to be merged.

My worry is that reviewers in my experience tend to enforce on others their view of how the code should be written - they end up becoming particular about how "they would write this code" and I think it's a harmful approach.
I've seen reviewers blocking on use of loops vs. closures or other quite arbitrary adjustments that don't have real effect on the performance or memory characteristics because of how they like to write the code.

My intention with this paragraph is to encourage the culture of trust in patch author skills and take a more conservative role as a reviewer - flag what's wrong, and let the patch author be right in everything else.

Can you suggest a paragraph that would better express that model while recognizing your position?

Reply:
If there's a disagreement over "loops vs. closures", that's a question that should be bubbled up to the SC and resolved by a change to the style guide.

> Can you suggest a paragraph that would better express that model while recognizing your position?

Maybe something like,

> An important piece of the reviewer's role is to correctly employ the three types of review comments (required, suggestion, or optional).

I would also consider merging this paragraph with the paragraph above.